apps/smart-contracts/core/test/DepositTradeHelper.test.ts

```diff
@@ -203,5 +252,100 @@ describe('=> DepositTradeHelper', () => {
         )
       })
     })
+
+    describe('testing with approvals already given', () => {
+      let baseTokenPermit: PermitStruct
+      let collateralPermit: PermitStruct
+      let timestampToSignFor: number
+      beforeEach(async () => {
+        timestampToSignFor = (await getLastTimestamp(ethers.provider)) + 5
+        baseTokenPermit = await getPermitFromSignature(
+          baseToken,
+          user,
+          depositTradeHelper.address,
+          ethers.constants.MaxUint256,
+          timestampToSignFor
+        )
+        collateralPermit = await getPermitFromSignature(
+          collateral,
+          user,
+          depositTradeHelper.address,
+          ethers.constants.MaxUint256,
+          timestampToSignFor
+        )
+      })
+
+      it('reverts if insufficient base token', async () => {
+        const userBTBalanceBefore = await baseToken.balanceOf(user.address)
+        await setNextTimestamp(ethers.provider, timestampToSignFor)
+
+        await expect(
+          depositTradeHelper
+            .connect(user)
+            .depositAndTrade(
+              userBTBalanceBefore.add(1),
+              baseTokenPermit,
+              collateralPermit,
+              junkTradeParams
+            )
+        ).to.be.revertedWith('ERC20: transfer amount exceeds balance')
+      })
+
+      it('takes `baseTokenAmount` from user prior to minting Collateral', async () => {
+        const userBTBalanceBefore = await baseToken.balanceOf(user.address)
+        await setNextTimestamp(ethers.provider, timestampToSignFor)
+
+        await depositTradeHelper
+          .connect(user)
+          .depositAndTrade(baseTokenToDeposit, baseTokenPermit, collateralPermit, junkTradeParams)
+
+        expect(await baseToken.balanceOf(user.address)).to.be.eq(
+          userBTBalanceBefore.sub(baseTokenToDeposit)
+        )
+        expect(baseToken.transferFrom.atCall(0)).to.have.been.calledWith(
```

Comment:
nit:
```suggestion
        expect(baseToken.transferFrom.atCall(0)).calledWith(
```
same/similarly for others